### PR TITLE
[release/7.0.4xx] support correct property name and ensure compat with old name

### PR DESCRIPTION
**Pull request by baronfel**
Closes https://github.com/dotnet/sdk-container-builds/issues/124 and paves the way for https://github.com/dotnet/sdk-container-builds/issues/381.

Renames `ContainerImageName` to `ContainerRepository` in the props and documentation, but maintains the use of `ContainerImageName` if present. Usage of `ContainerImageName` fires off a warning as well instructing the user to use `ContainerRepository` instead.

### CODE REVIEW

**Review by MichalPavlik — APPROVED** [review]

File: src/Containers/Microsoft.NET.Build.Containers/KnownStrings.cs [hunk 1]
```diff
@@ -31,6 +33,8 @@ public static class Properties
 
     public static class ErrorCodes
     {
+        public static readonly string CONTAINER002 = nameof(CONTAINER002);
+        public static readonly string CONTAINER003 = nameof(CONTAINER003);
```
Comment (vlada-shubina) [inline src/Containers/Microsoft.NET.Build.Containers/KnownStrings.cs:37]: We have started all the codes from 10xx, is it intentional?

File: src/Containers/Microsoft.NET.Build.Containers/PublicAPI/net472/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -47,8 +47,8 @@ Microsoft.NET.Build.Containers.Tasks.CreateNewImage.GeneratedContainerConfigurat
 Microsoft.NET.Build.Containers.Tasks.CreateNewImage.GeneratedContainerConfiguration.set -> void
 Microsoft.NET.Build.Containers.Tasks.CreateNewImage.GeneratedContainerManifest.get -> string!
 Microsoft.NET.Build.Containers.Tasks.CreateNewImage.GeneratedContainerManifest.set -> void
-Microsoft.NET.Build.Containers.Tasks.CreateNewImage.ImageName.get -> string!
-Microsoft.NET.Build.Containers.Tasks.CreateNewImage.ImageName.set -> void
+Microsoft.NET.Build.Containers.Tasks.CreateNewImage.Repository.get -> string!
+Microsoft.NET.Build.Containers.Tasks.CreateNewImage.Repository.set -> void
```
Comment (vlada-shubina) [inline src/Containers/Microsoft.NET.Build.Containers/PublicAPI/net472/PublicAPI.Unshipped.txt:51]: `ContainerRepository`, just to stay consistent?
also do we need to change `BaseImageName`?

File: src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/FluentAssertionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+}
+
+public static class ProjectInstanceExtensions
+{
+    public static ProjectInstanceAssertions Should(this ProjectInstance instance)
+    {
+        return new ProjectInstanceAssertions(instance);
+    }
+}
```
Comment (vlada-shubina) [inline src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/FluentAssertionExtensions.cs:42]: should we add it to `*.NET.TestFramework` instead?

**Review by vlada-shubina — CHANGES_REQUESTED** [review]
Do we need to keep some tests to ensure old name is still supported?

File: src/Containers/Microsoft.NET.Build.Containers/KnownStrings.cs (hunk 1 above)
Comment (baronfel) [reply src/Containers/Microsoft.NET.Build.Containers/KnownStrings.cs:37]: The diagnostics emitted directly from the targets are still three-digits and so I just kept on with that for compat. We could break this now if we wanted.

File: src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/FluentAssertionExtensions.cs (hunk 3 above)
Comment (baronfel) [reply src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/FluentAssertionExtensions.cs]: This wasn't actually used after some refactoring so I deleted it.

File: src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/TargetsTests.cs [hunk 4]
```diff
@@ -31,6 +31,21 @@ public void CanSetEntrypointArgsToUseAppHost(bool useAppHost, params string[] en
...
+        {
+            [ContainerImageName] = customImageName
+        });
+        using var _ = d;
+        var instance = project.CreateProjectInstance(global::Microsoft.Build.Execution.ProjectInstanceSettings.None);
+        instance.Build(new[] { ComputeContainerConfig }, new []{ logger });
+        logger.Warnings.Should().HaveCount(1, "a warning for the use of the old ContainerImageName property should have been created");
+        logger.Warnings[0].Code.Should().Be(KnownStrings.ErrorCodes.CONTAINER003);
+        Assert.Equal(customImageName, instance.GetPropertyValue(ContainerRepository));
+    }
```
Comment (baronfel) [inline src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/TargetsTests.cs:47]: @vlada-shubina here is the test that confirms that we still use the previous property - just with a warning that lets users know to migrate

**Review by vlada-shubina — APPROVED** [review]

**Pull request merged**